### PR TITLE
Issue#1264: Implemented Composite-View Pattern

### PR DESCRIPTION
**Pull request by Kevinyl3**
Issue#1264: Composite-View Pattern

- Implemented an example of the Composite-View pattern 
- Servlet runs on Tomcat 10+, uses JSP to generate views

Pull request description

- created `composite-view` directory/module for the pattern 
- `src\main` contains the javabean class for view management, and the Servlet class
- `src\test` contains the tests for the methods in the javabean class
- `web` contains the all the JSP files and the web.xml file

### CODE REVIEW

File: composite-view/README.md [hunk 1]
```diff
@@ -0,0 +1,321 @@
+---
+layout: pattern
+title: Composite View 
+folder: composite-view
+permalink: /patterns/composite-view/
+categories: structural
+language: en
+tags:
+- Enterprise Integration Pattern
```
Comment (iluwatar) [inline composite-view/README.md:9]: We should add `Presentation` tag

File: composite-view/README.md [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+---
+
+## Name
+**Composite View Pattern**
```
Comment (iluwatar) [inline composite-view/README.md:13]:
```suggestion
Composite View
```

File: composite-view/README.md [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+> Use composite views that are composed of multiple atomic subviews. Each component of 
+> the template may be included dynamically into the whole and the layout of the page may be managed independently of the content.
+> This solution provides for the creation of a composite view based on the inclusion and substitution of 
+> modular dynamic and static template fragments. 
+> It promotes the reuse of atomic portions of the view by encouraging modular design.
+
+**Programmatic Example**
+
+Since this is a web development pattern, we will need to set up a server for this example.
+I chose to use Tomcat 10.0.13 for my case, and this programmatic example will only work with Tomcat 10+.
```
Comment (iluwatar) [inline composite-view/README.md:42]: Please use passive voice https://www.grammarly.com/blog/passive-voice/

Comment (iluwatar) [reply composite-view/README.md:42]: Also, run the `README.md` through a spell-checker such as grammarly.com

File: composite-view/README.md [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+* Code is more maintainable and modular
+
+**Cons**
+* Overhead cost at runtime
+* Slower response compared to directly embedding elements
+* Increases potential for display errors 
+
+## Related patterns
+* Composite (GoF)
+* View Helper
```
Comment (iluwatar) [inline composite-view/README.md:316]: Add links. At least Composite is part of this repository already.

File: composite-view/pom.xml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            <groupId>jakarta.servlet</groupId>
+            <artifactId>jakarta.servlet-api</artifactId>
+            <version>5.0.0</version>
+            <scope>compile</scope>
+        </dependency>
+    </dependencies>
+    <properties>
+        <maven.compiler.source>16</maven.compiler.source>
+        <maven.compiler.target>16</maven.compiler.target>
+    </properties>
```
Comment (iluwatar) [inline composite-view/pom.xml:57]: Not needed. We use the target version coming from the parent `pom.xml`.

File: composite-view/src/main/java/com/iluwatar/compositeview/ClientPropertiesBean.java [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+  }
+
+  /**
+   * Getter for name.
+   *
+   * @return the name string
+   */
+  public String getName() {
+    return name;
+  }
```
Comment (iluwatar) [inline composite-view/src/main/java/com/iluwatar/compositeview/ClientPropertiesBean.java:68]: Use Lombok to get rid of boilerplate

File: composite-view/src/test/java/com/illuwatar/compositeview/JavaBeansTest.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
+package com.illuwatar.compositeview;
```
Comment (iluwatar) [inline composite-view/src/test/java/com/illuwatar/compositeview/JavaBeansTest.java:1]: The file path should be `composite-view/src/test/java/com/iluwatar/compositeview/JavaBeansTest.java`

**Review by iluwatar — CHANGES_REQUESTED** [review]

**Comment by iluwatar** [comment]
In addition, SonarCloud code analysis is failing due to insufficient test coverage. Try to add more tests.

File: composite-view/src/test/java/com/illuwatar/compositeview/JavaBeansTest.java (hunk 7 above)
Comment (Kevinyl3) [reply composite-view/src/test/java/com/illuwatar/compositeview/JavaBeansTest.java:1]: That is the file path I have right now.

**Comment by Kevinyl3** [comment]
for the code coverage, I'm not sure what tools/libraries there are to unit test HTTP servlets, do you have any suggestions?

File: composite-view/src/test/java/com/illuwatar/compositeview/JavaBeansTest.java (hunk 7 above)
Comment (tedyoung) [reply composite-view/src/test/java/com/illuwatar/compositeview/JavaBeansTest.java:1]: Tests are in the wrong package/directory name. One "el" not two.

> `composite-view/src/test/java/com/illuwatar/compositeview/JavaBeansTest.java`

should be

> `composite-view/src/test/java/com/iluwatar/compositeview/JavaBeansTest.java`

and package should be `com.iluwatar.compositevie` not `com.illuwatar.compositeview`

File: composite-view/src/test/java/com/illuwatar/compositeview/JavaBeansTest.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        assertEquals("DEFAULT_NAME", newBean.getName());
+        assertTrue(newBean.isBusinessInterest());
+        assertTrue(newBean.isScienceNewsInterest());
+        assertTrue(newBean.isSportsInterest());
+        assertTrue(newBean.isWorldNewsInterest());
+
+    }
+
+    @Test
+    public void testNameGetterSetter() {
```
Comment (tedyoung) [inline composite-view/src/test/java/com/illuwatar/compositeview/JavaBeansTest.java:21]: Are tests for generated setter/getters really necessary?

File: composite-view/src/test/java/com/illuwatar/compositeview/JavaBeansTest.java (hunk 7 above)
Comment (Kevinyl3) [reply composite-view/src/test/java/com/illuwatar/compositeview/JavaBeansTest.java:1]: ah I see, thanks

File: composite-view/src/test/java/com/illuwatar/compositeview/JavaBeansTest.java (hunk 8 above)
Comment (Kevinyl3) [reply composite-view/src/test/java/com/illuwatar/compositeview/JavaBeansTest.java:21]: I will write some mocking tests for the AppServer and the non-empty constructor, but if I remove these getter/setter tests, I think the sonar cloud analysis will still fail, due to not enough line coverage.

File: composite-view/README.md (hunk 1 above)
Comment (Kevinyl3) [reply composite-view/README.md:9]: will do, I will implement these suggestions.

File: composite-view/README.md [hunk 9]
```diff
@@ -0,0 +1,325 @@
+---
+layout: pattern
+title: Composite View 
+folder: composite-view
+permalink: /patterns/composite-view/
+categories: structural
```
Comment (iluwatar) [inline composite-view/README.md:6]:
```suggestion
categories: Structural
```

File: composite-view/README.md [hunk 10]
```diff
@@ -0,0 +1,325 @@
...
+* Reduce code duplication
+* Code is more maintainable and modular
+
+**Cons**
+* Overhead cost at runtime
+* Slower response compared to directly embedding elements
+* Increases potential for display errors 
+
+## Related patterns
+* [Composite (GoF)](../composite)
```
Comment (iluwatar) [inline composite-view/README.md:319]: I suggest linking to the website https://java-design-patterns.com/patterns/composite/

**Review by iluwatar — CHANGES_REQUESTED** [review]

File: composite-view/README.md (hunk 10 above)
Comment (Kevinyl3) [reply composite-view/README.md]: apologies, I was busy for the holidays, I have added these changes in my latest commit.

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Excellent, thank you for adding this pattern!
@all-contributors please add @Kevinyl3 for code

**Pull request merged**